### PR TITLE
Add command to restart LSP server

### PR DESCRIPTION
**Pull request by threecgreen**
Implements #523. 

Based on the discussion in #533, there's additional events I should be handling, but I wanted to make sure I was on the right track.

### CODE REVIEW

File: helix-lsp/src/lib.rs [hunk 1]
```diff
@@ -343,52 +343,81 @@ impl Registry {
...
+        if let Some((_, client)) = self.inner.get(&language_config.scope) {
+            Ok(client.clone())
+        } else {
+            let id = self.counter.fetch_add(1, Ordering::Relaxed);
+            let client = self.initialize_client(language_config, config, id)?; // initialize a new client
+            self.inner
+                .insert(language_config.scope.clone(), (id, client.clone()));
+            Ok(client)
+        }
+    }
```
Comment (archseer) [inline helix-lsp/src/lib.rs:355]: Why the change here? This was deliberate since we don't want to try reinitializing the client if we already tried starting it and failed.

Comment (threecgreen) [reply helix-lsp/src/lib.rs:355]: the functionality of `get` should be unchanged; I pulled out `initialize_client` into its own function so that I could also use it in `restart`. I switched from using `self.inner.entry` to `self.inner.get` to avoid multiple mutable borrows.

**Comment by Niedzwiedzw** [comment]
this!!!!
